master/internal/kubernetes/spec.go

```diff
@@ -175,16 +181,27 @@ func (p *pod) configureVolumes(
 }
 
 func (p *pod) modifyPodSpec(newPod *k8sV1.Pod, scheduler string) {
-	if scheduler != "coscheduler" || newPod.Spec.SchedulerName != "" ||
-		p.taskSpec.Description() == "cmd" {
+	if p.taskSpec.Description() == cmdTask {
+		return
+	}
+
+	if scheduler == coscheduler || scheduler == preemptionScheduler {
```

Comment:
I think the refactor is WAY clearer - thanks. I hoped Github would show this as a "suggestion" as part of a review (a-la Google Docs), but it looks like it literally added a commit. What do you think of just combining the 2 conditionals for the 2 scheduler types? For now they're identical, the conditional is still super clear (IMO anyway), and this reduces the code a little bit. If you're agreeable, I really like this.